floodmodeller_api/test/test_toolbox_structure_log.py

```diff
@@ -79,156 +83,110 @@ def structure():
-    output = slb._section_data(conduit_empty, no_length)
-    assert output == ["Colebrook-White: 0", "h: 0.00 x w: 0.00 x l: 0.00"]
-    output = slb._section_data(conduit_filled, with_length)
-    assert output == [
-        "Colebrook-White: [min: 0.0, max: 4.0]",
-        "h: 65.00 x w: 150.00 x l: 4.97",
-    ]
-
-
-def test_sprung_data(slb, conduit_empty, conduit_filled, no_length, with_length):
-    output = slb._sprung_data(conduit_empty, no_length)
-    assert output == [
-        "Mannings: 0",
-        "(Springing: 0.00, Crown: 0.00) x w: 0.00 x l: 0.00",
-    ]
-    output = slb._sprung_data(conduit_filled, with_length)
-    assert output == [
-        "Mannings: [min: 1.34, max: 1.876]",
-        "(Springing: 23.10, Crown: 5.40) x w: 3.00 x l: 4.97",
-    ]
-
-
-def test_orifice_dimensions(slb, structure):
-    structure.invert = 1
-    output = slb._orifice_dimensions(structure)
-    assert output == "h: -1.00 x w: -1.00"
-
-
-def test_spill_data(slb, structure):
-    structure.data = pd.DataFrame(data={"X": [0, 0], "Y": [0, 0]})
-    structure.weir_coefficient = 0
-    output = slb._spill_data(structure)
-    assert output == ["Elevation: 0.00 x w: 0.00", 0]
-
-
-def test_bridge_data(slb, structure):
-    structure.section_data = pd.DataFrame(data={"X": [0, 0], "Y": [0, 0], "Mannings n": [0, 0]})
-    structure.opening_data = pd.DataFrame(
-        data={"Start": 0, "Finish": 0, "Springing Level": 0, "Soffit Level": 0},
-        index=[0],
-    )
-    output = slb._bridge_data(structure)
-    assert output == ["Mannings: 0", "h: 0.00 x w: 0.00"]
+@pytest.fixture
+def ex18_dat_expected():
+    # I think this is the correct way to establish the same 'expected' result, but if this is a larger/more structure-rich dat, could read from file?
+    # Ideally I'd use a larger dat for this like one of the ones from private sample-dataset, to be discussed
+    return """Unit Name,Unit Type,Unit Subtype,Comment,Friction,Dimensions (m),Weir Coefficient,Culvert Inlet/Outlet Loss
+C2,CONDUIT,CIRCULAR,,"Mannings: [min: 0.015, max: 0.020]",dia: 1.00 x l: 100.00 (Total conduit length: 500.00),,Ki: 0.6
+C2_R1,REPLICATE,,,,,,
+C2_R2,REPLICATE,,,,,,
+C2_R3,REPLICATE,,,,,,
+C2_R4,REPLICATE,,,,,,
+C2m,CONDUIT,CIRCULAR,,"Mannings: [min: 0.015, max: 0.020]",dia: 1.00 x l: 0.00,,
+C2md,CONDUIT,CIRCULAR,,"Mannings: [min: 0.015, max: 0.020]",dia: 1.00 x l: 100.00 (Total conduit length: 700.00),,
+C2_R5,REPLICATE,,,,,,
+C2_R6,REPLICATE,,,,,,
+C2_R7,REPLICATE,,,,,,
+C2_R8,REPLICATE,,,,,,
+C2_R9,REPLICATE,,,,,,
+C2_R10,REPLICATE,,,,,,
+C2d,CONDUIT,CIRCULAR,,"Mannings: [min: 0.015, max: 0.020]",dia: 1.00 x l: 0.00,,
+S0,WEIR,,,,Crest Elevation: 21.00 x w: 1.50,,
+C2d,WEIR,,,,Crest Elevation: 18.00 x w: 0.60,,
+S4,WEIR,,,,Crest Elevation: 17.90 x w: 2.00,,
+S8,WEIR,,,,Crest Elevation: 17.70 x w: 2.00,,
+S3LS,SPILL,,,,Elevation: 20.00 x w: 100.00,1.7,
+"""
 
 
-def test_add_conduits(slb, conduit_filled, tmpdir):
+def test_conduit_data(slb, conduit_empty):
```

Comment:
I would maybe rename this to something like `test_empty_conduit` so it's more descriptive

Reply:
addressed